pycloudlib/ec2/instance.py

```diff
@@ -81,9 +105,67 @@ def add_network_interface(self) -> str:
         See the AWS documentation for more info:
         https://boto3.readthedocs.io/en/latest/reference/services/ec2.html?#EC2.Client.create_network_interface
         https://boto3.readthedocs.io/en/latest/reference/services/ec2.html?#EC2.Client.attach_network_interface
+
+        Args:
+            ipv4_address_count: number of private ipv4s
+            ipv6_address_count: number of private ipv6s
+            ipv4_public_ip_count: number of public ips associated to private
+                ips.
+            associate_public_ip: created boto3 instance object
+        Returns: private primary ip
         """
         self._log.debug("adding network interface to %s", self.id)
-        interface_id = self._create_network_interface()
+        interface_id = self._create_network_interface(ipv6_address_count)
+        if ipv4_public_ip_count is not None:
+            if (
+                ipv4_address_count is None
+                or ipv4_address_count < ipv4_public_ip_count
+            ):
+                raise PycloudlibError(
+                    f"Invalid combination of"
+                    f" `ipv4_address_count={ipv4_public_ip_count}` and"
```

Comment:
I think you meant `ipv4_public_ip_count`.

Reply:
Good catch, thanks. Updated.